### PR TITLE
Feature: added remote network `location` property

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/231

## Changes
- added `location` property to remote-network resource and datasource

### CODE REVIEW

File: twingate/internal/provider/datasource/remote-network.go [hunk 1]
```diff
@@ -44,6 +44,11 @@ func RemoteNetwork() *schema.Resource {
 				Description:  "The name of the Remote Network",
 				ExactlyOneOf: []string{"id"},
 			},
+			"location": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The location of the Remote Network",
```
Comment (vmanilo) [inline twingate/internal/provider/datasource/remote-network.go:50]: @alexmensch please check description

File: twingate/internal/provider/resource/remote-network.go [hunk 2]
```diff
@@ -30,6 +31,13 @@ func RemoteNetwork() *schema.Resource {
 				Required:    true,
 				Description: "The name of the Remote Network",
 			},
+			"location": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(model.Locations, false),
+				Description:  "The location of the Remote Network",
```
Comment (vmanilo) [inline twingate/internal/provider/resource/remote-network.go:38]: @alexmensch please check description

File: twingate/internal/model/remote-network.go [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
 
+const (
+	LocationAWS         = "AWS"
+	LocationAzure       = "AZURE"
+	LocationGoogleCloud = "GOOGLE_CLOUD"
+	LocationOnPremise   = "ON_PREMISE"
+	LocationOther       = "OTHER"
+)
+
+var Locations = []string{LocationAWS, LocationAzure, LocationGoogleCloud, LocationOnPremise, LocationOther} //nolint
```
Comment (ekampf) [inline twingate/internal/model/remote-network.go:11]: What happens if we add a new Enum value on server side?   will it break TF?

Comment (vmanilo) [reply twingate/internal/model/remote-network.go:11]: new enum value won’t break it
* datasource accepts any value as they fetched from backend side (so we don’t need to validate it)
* on resource side - user should provide this value and terraform validates according to our list of possible locations, so when new enum added on server side - terraform just won’t allow to use this new value until we update list of allowed location options

File: twingate/internal/provider/datasource/remote-network.go (hunk 1 above)
Comment (alexmensch) [reply twingate/internal/provider/datasource/remote-network.go:50]: We should list out the possible options here. So let's change it to: `The location of the Remote Network. Must be one of the following: AWS, AZURE, GOOGLE_CLOUD, ON_PREMISE, OTHER`

File: twingate/internal/provider/resource/remote-network.go (hunk 2 above)
Comment (alexmensch) [reply twingate/internal/provider/resource/remote-network.go:38]: We should list out the possible options here. So let's change it to: The location of the Remote Network. Must be one of the following: AWS, AZURE, GOOGLE_CLOUD, ON_PREMISE, OTHER

File: twingate/internal/provider/datasource/remote-network.go (hunk 1 above)
Comment (alexmensch) [reply twingate/internal/provider/datasource/remote-network.go:50]: The only downside to this is that the docs aren't automatically up to date with our API, but I don't think these are going to change very often.

**Review by alexmensch — CHANGES_REQUESTED** [review]
Just need to provide a list of what the available options are somewhere.

**Review by alexmensch — APPROVED** [review]
Nice, I like your solution to list them out programmatically :)

**Review by ekampf — APPROVED** [review]

**Pull request merged**